homework3/src/main/java/HostDevice.java

```diff
@@ -0,0 +1,65 @@
+public class HostDevice {
+
+
+    public ElectricityConsumer[] consumers = new ElectricityConsumer[10];
+    public SmartDevice[] device = new SmartDevice[10];
+    SmartLamp s = new SmartLamp();
+
+
+    public HostDevice(){
+            consumers[0] = new SmartLamp();
+            consumers[1] = new SmartThermostat();
+            consumers[2] = new SmartSecurityCamera();
+
+            device[0]= (SmartDevice) new SmartLamp();
```

Comment:
do you need this explicit cast? Is this a down or up-cast? Where do we need explicit casting?